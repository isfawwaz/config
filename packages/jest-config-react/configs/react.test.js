const withReact = require('./react');

const buildConfig = () => ({
  testEnvironment: 'jsdom',
  transform: {
    foo: 'bar',
  },
  moduleNameMapper: {
    foz: 'baz',
  },
});

it('should match snapshot', () => {
  expect(withReact()(buildConfig())).toMatchSnapshot();
});

it('should throw if testEnvironment is not jsdom', () => {
  const config = {
    ...buildConfig(),
    testEnvironment: 'node',
  };

  expect(() => withReact()(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
});
