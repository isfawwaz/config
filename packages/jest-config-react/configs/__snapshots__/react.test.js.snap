// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "moduleNameMapper": {
    "/.css$": "<HOME_DIR>/Programming/web/self-dev/personal-config/node_modules/.pnpm/identity-obj-proxy@3.0.0/node_modules/identity-obj-proxy/src/index.js",
    "foz": "baz",
  },
  "testEnvironment": "jsdom",
  "transform": {
    "foo": "bar",
  },
}
`;
